source: http://permalink.gmane.org/gmane.comp.file-systems.fuse.devel/9570
Upstream: still no as of 2.8.5
Reason: linking fails with gold
diff --git a/fuse_2.7.4/lib/fuse.c b/fuse-2.7.4/lib/fuse.c
index fc4d967..a98cd59 100644
--- a/fuse_2.7.4/lib/fuse.c
+++ b/fuse-2.7.4/lib/fuse.c
@@ -3330,11 +3330,11 @@ struct fuse *fuse_new_compat1(int fd, int flags,
 				      11);
 }
 
-FUSE_SYMVER(".symver fuse_exited,__fuse_exited@");
-FUSE_SYMVER(".symver fuse_process_cmd,__fuse_process_cmd@");
-FUSE_SYMVER(".symver fuse_read_cmd,__fuse_read_cmd@");
-FUSE_SYMVER(".symver fuse_set_getcontext_func,__fuse_set_getcontext_func@");
-FUSE_SYMVER(".symver fuse_new_compat2,fuse_new@");
+FUSE_SYMVER(".symver fuse_exited,__fuse_exited@FUSE_UNVERSIONED");
+FUSE_SYMVER(".symver fuse_process_cmd,__fuse_process_cmd@FUSE_UNVERSIONED");
+FUSE_SYMVER(".symver fuse_read_cmd,__fuse_read_cmd@FUSE_UNVERSIONED");
+FUSE_SYMVER(".symver fuse_set_getcontext_func,__fuse_set_getcontext_func@FUSE_UNVERSIONED");
+FUSE_SYMVER(".symver fuse_new_compat2,fuse_new@FUSE_UNVERSIONED");
 FUSE_SYMVER(".symver fuse_new_compat22,fuse_new@FUSE_2.2");
 
 #endif /* __FreeBSD__ */
diff --git a/fuse_2.7.4/lib/fuse_mt.c b/fuse-2.7.4/lib/fuse_mt.c
index 95c3a5c..4ffaf40 100644
--- a/fuse_2.7.4/lib/fuse_mt.c
+++ b/fuse-2.7.4/lib/fuse_mt.c
@@ -113,4 +113,4 @@ int fuse_loop_mt(struct fuse *f)
 	return fuse_session_loop_mt(fuse_get_session(f));
 }
 
-FUSE_SYMVER(".symver fuse_loop_mt_proc,__fuse_loop_mt@");
+FUSE_SYMVER(".symver fuse_loop_mt_proc,__fuse_loop_mt@FUSE_UNVERSIONED");
diff --git a/fuse_2.7.4/lib/fuse_session.c b/fuse-2.7.4/lib/fuse_session.c
index 5efedd9..967740a 100644
--- a/fuse_2.7.4/lib/fuse_session.c
+++ b/fuse-2.7.4/lib/fuse_session.c
@@ -207,4 +207,6 @@ void fuse_chan_destroy(struct fuse_chan *ch)
 
 #ifndef __FreeBSD__
 FUSE_SYMVER(".symver fuse_chan_new_compat24,fuse_chan_new@FUSE_2.4");
+#else
+FUSE_SYMVER(".symver fuse_chan_new,fuse_chan_new@FUSE_2.4");
 #endif
diff --git a/fuse_2.7.4/lib/fuse_versionscript b/fuse-2.7.4/lib/fuse_versionscript
index 3eedd0c..bed94d6 100644
--- a/fuse_2.7.4/lib/fuse_versionscript
+++ b/fuse-2.7.4/lib/fuse_versionscript
@@ -1,3 +1,6 @@
+FUSE_UNVERSIONED {
+};
+
 FUSE_2.2 {
 	global:
 		fuse_destroy;
@@ -28,7 +31,6 @@ FUSE_2.4 {
 		fuse_chan_data;
 		fuse_chan_destroy;
 		fuse_chan_fd;
-		fuse_chan_new;
 		fuse_chan_receive;
 		fuse_chan_send;
 		fuse_chan_session;
diff --git a/fuse_2.7.4/lib/helper.c b/fuse-2.7.4/lib/helper.c
index e3c5985..76dfd05 100644
--- a/fuse_2.7.4/lib/helper.c
+++ b/fuse-2.7.4/lib/helper.c
@@ -408,10 +408,10 @@ int fuse_mount_compat1(const char *mountpoint, const char *args[])
 	return fuse_mount_compat22(mountpoint, NULL);
 }
 
-FUSE_SYMVER(".symver fuse_setup_compat2,__fuse_setup@");
+FUSE_SYMVER(".symver fuse_setup_compat2,__fuse_setup@FUSE_UNVERSIONED");
 FUSE_SYMVER(".symver fuse_setup_compat22,fuse_setup@FUSE_2.2");
-FUSE_SYMVER(".symver fuse_teardown,__fuse_teardown@");
-FUSE_SYMVER(".symver fuse_main_compat2,fuse_main@");
+FUSE_SYMVER(".symver fuse_teardown,__fuse_teardown@FUSE_UNVERSIONED");
+FUSE_SYMVER(".symver fuse_main_compat2,fuse_main@FUSE_UNVERSIONED");
 FUSE_SYMVER(".symver fuse_main_real_compat22,fuse_main_real@FUSE_2.2");
 
 #endif /* __FreeBSD__ */
